Commons/bungee/src/cl/bgmp/bungee/commands/staffchat/StaffChatManager.java

```diff
@@ -0,0 +1,92 @@
+package cl.bgmp.bungee.commands.staffchat;
+
+import cl.bgmp.bungee.BungeeMessages;
+import cl.bgmp.bungee.ChatConstant;
+import cl.bgmp.bungee.ChatState;
+import cl.bgmp.bungee.CommonsBungee;
+import cl.bgmp.bungee.Permission;
+import cl.bgmp.bungee.Util;
+import java.util.HashMap;
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.chat.ComponentBuilder;
+import net.md_5.bungee.api.chat.TextComponent;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.api.event.ChatEvent;
+import net.md_5.bungee.api.event.ServerConnectEvent;
+import net.md_5.bungee.api.plugin.Listener;
+import net.md_5.bungee.event.EventHandler;
+import org.jetbrains.annotations.NotNull;
+
+public class StaffChatManager implements Listener {
+
+  public static HashMap<String, ChatState> getPlayerChatStates() {
+    return CommonsBungee.getPlayerChatStates();
+  }
+
+  public static void alternatePlayerChat(ProxiedPlayer player) {
+    final ChatState alternation =
+        getPlayerChatStates().get(player.getName()) == ChatState.GLOBAL
+            ? ChatState.STAFF
+            : ChatState.GLOBAL;
+    final ChatConstant message =
+        alternation == ChatState.GLOBAL
+            ? ChatConstant.LOCKED_GLOBAL_CHAT
+            : ChatConstant.LOCKED_STAFF_CHAT;
+
+    getPlayerChatStates().put(player.getName(), alternation);
+    player.sendMessage(
+        BungeeMessages.append(
+            ChatConstant.STAFF_CHAT_PREFIX.getAsTextComponent(), message.getAsString()));
+  }
```

Comment:
This doesn't really belong in `StaffChatManager`, but if you only have 2 chat types it should work i guess. Looks like a future refactor though if you want to add more channels in the future

Reply:
Very unlikely that I will add more channels. And yes, given the current logic, it does make sense, and will be refactored if the aim of this changes and more `ChatState`s pop up.